gcstorage/gcstorage_test.go

```diff
@@ -0,0 +1,184 @@
+package gcstorage
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+var emulator *exec.Cmd
+
+func setup(t *testing.T) {
+	_ = exec.Command(`docker`, `rm`, `-f`, `gcp-storage-emulator`).Run()
+
+	err := exec.Command(`docker`, `pull`, `oittaa/gcp-storage-emulator`).Run()
+	require.NoError(t, err)
+
+	emulator = exec.Command(`docker`, `run`, `-e`, `PORT=9023`, `-p`, `9023:9023`, `--name`, `gcp-storage-emulator`,
+		`oittaa/gcp-storage-emulator`, `start`, `--port=9023`, `--in-memory`, `--default-bucket=my-bucket`)
```

Comment:
nit: no need for `docker rm` / `docker pull`

I think we can `docker run` only and defer `docker rm -f`

Reply:
Its a setup function. It can not remove emulator before tests are done. It's being removed in ``teardown``.

Reply:
Also, it's hard to estimate how long it will take to pull the image. The remove is there in case it has not be removed for some reason in a previous test run e.g. crash...